.travis.yml

```diff
@@ -19,19 +19,35 @@ matrix:
       before_install: source script/travis/install_qt5.sh
     - os: linux
       env: STATIC_ANALYSIS=ON
+    - os: linux
+      env: DYNAMIC_ANALYSIS=ON CMAKE_OPTIONS="-DCMAKE_BUILD_TYPE=Debug"
+      addons:
+        apt:
+          update: true
+          packages: valgrind
     - os: osx
       env: QT_BASE=latest CMAKE_OPTIONS="-DPENGUINV_BUILD_QT_EXAMPLE=ON"
       before_install: source script/travis/install_qt5.sh
 
 script:
-  - if [[ "$STATIC_ANALYSIS" == "ON" ]]; then
+  - |
+    if [[ "$STATIC_ANALYSIS" == "ON" ]]; then
       export EXCLUDE_PATTERN_1=" not found in expected header ";
       export EXCLUDE_PATTERN_2=" does not need to be \#included; use a forward declaration instead";
       export EXCLUDE_PATTERN_3=" not found in any directly \#included header";
       git clone https://github.com/myint/cppclean;
       ./cppclean/cppclean src test examples | grep -v "$EXCLUDE_PATTERN_1" | grep -v "$EXCLUDE_PATTERN_2" | grep -v "$EXCLUDE_PATTERN_3";
       exit 0;
```

Comment:
See for static analysis we don't execute `after_script` too and we have the same command here.

Reply:
You're fully right. But without `exit`, the subsequent commands will be executed. I must come up with a better logic.

Reply:
You could try something like this:
```bash
  - mkdir build
  - cd build
  - cmake .. ${CMAKE_OPTIONS}
  - |
    if [[ "$DYNAMIC_ANALYSIS" == "ON" ]]; then
      mkdir -p build && cd build
      cmake .. ${CMAKE_OPTIONS}
      cmake --build . --target unit_tests -- -j2
      ctest -D ExperimentalBuild -j2
      ctest -E perf_test --extra-verbose -T MemCheck -j2
    else
      cmake --build .
       ctest -E perf_test --extra-verbose
    fi
```

`CUDA` doesn't need changes